src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -315,12 +317,19 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
   };
 
   const getNurseriesOptions = () => {
-    const nurseries = batches.reduce((acc, batch) => {
-      if (!acc[batch.facility_id.toString()]) {
-        acc[batch.facility_id.toString()] = { label: batch.facility_name, value: batch.facility_id };
-      }
-      return acc;
-    }, {});
+    const nurseries = batches
+      .filter((batch) => {
+        if (isOutplant) {
+          return +batch.readyQuantity > 0;
+        }
+        return +batch.totalQuantity > 0;
+      })
+      .reduce((acc, batch) => {
+        if (!acc[batch.facility_id.toString()]) {
```

Comment:
I think we could include the filter condition above in this same reduce:

`  if ((!isOutplant || +batch.readyQuantity > 0) && !acc[batch.facility_id.toString()]) {`

(didn't tested, just thinking)

Reply:
That will also work. Being functional is easier to read, we should leverage functional programming more if we can. Unless performance is a concern.